Updated samples for new documentation (#62)
auth0-samples/auth0-vue-samples: 03-Calling-an-API/README.md

# Scenario #3 - Calling an External API

For this scenario, an API endpoint `/api/external` has been included in the Express server that requires a bearer token to be supplied as a bearer token in the `Authorization` header (as provided during the authentication flow). This uses the [`express-jwt`](https://github.com/auth0/express-jwt) middleware to validate the token against the identifier of your API as set up in the Auth0 dashboard, as well as checking that the signature is valid.

## Configuration

The project needs to be configured with your Auth0 domain and client ID in order for the authentication flow to work. Both the front end and the back end need to be configured.

To do this, first copy `src/auth/auth0-variables.sample.js` into a new file in the same folder called `auth0-variables.js`, and replace the values within with your own Auth0 application credentials.

```js
// src/auth/auth0-variables.js

export default {
  domain: "<YOUR AUTH0 DOMAIN>",

>   clientId: "<YOUR AUTH0 CLIENTID>",

Reviewer: ```suggestion
  clientId: "<YOUR AUTH0 CLIENT ID>",
```

  callbackUrl: "http://localhost:3000/callback",
  audience: process.env.AUTH0_AUDIENCE || "<YOUR AUTH0 AUDIENCE>"
};
```

> Next, copy `.env.sample` in the root of the project, and enter values for the `AUTH0_DOMAIN`, `AUTH0_CLIENTID` and `AUTH0_AUDIENCE` fields. These should be the same values that are used in the `auth0-variables.js` file:

Author: I agree, and this is something I tried to provide a solution for. I chatted with some others about this a good while ago now, and felt it was easiest to just provide two files, from the point of view of both configuration and explaining it to the reader.

What I would like to do is just provide one file with the configuration that can be used by both Node and Webpack. The difficulty at the moment is that Webpack uses ES6 modules whereas Node uses CommonJS, and they aren't compatible.

I will explore a solution to this (perhaps using a Json file) as long as it's not too cumbersome for the reader. The closer we can get it to working out-of-the-box, the better.

> Next, copy `.env.sample` in the root of the project, and enter values for the `AUTH0_DOMAIN`, `AUTH0_CLIENTID` and `AUTH0_AUDIENCE` fields. These should be the same values that are used in the `auth0-variables.js` file:

Author: This has now been changed and there is an `auth_config.json` file in the root of each project that is used by both the backend and frontend. CircleCI creates this file as part of its build step, too.

> Next, copy `.env.sample` in the root of the project, and enter values for the `AUTH0_DOMAIN`, `AUTH0_CLIENTID` and `AUTH0_AUDIENCE` fields. These should be the same values that are used in the `auth0-variables.js` file:

Reviewer: ```suggestion
Next, copy `.env.sample` in the root of the project, and enter values for the `AUTH0_DOMAIN`, `AUTH0_CLIENT_ID` and `AUTH0_AUDIENCE` fields. These should be the same values that are used in the `auth0-variables.js` file:
```

```bash
AUTH0_DOMAIN=<YOUR AUTH0 DOMAIN>

> AUTH0_CLIENTID=<YOUR AUTH0 CLIENT ID>

Reviewer: ```suggestion
AUTH0_CLIENT_ID=<YOUR AUTH0 CLIENT ID>
```
